common/static/common/js/discussion/views/discussion_thread_show_view.js

```diff
@@ -38,32 +38,34 @@
 
             DiscussionThreadShowView.prototype.renderTemplate = function() {
                 var context;
-                this.template = _.template($("#thread-show-template").html());
                 context = $.extend({
                     mode: this.mode,
                     flagged: this.model.isFlagged(),
                     author_display: this.getAuthorDisplay(),
                     cid: this.model.cid,
                     readOnly: $('.discussion-module').data('read-only')
                 }, this.model.attributes);
-                return this.template(context);
+                return edx.HtmlUtils.template($("#thread-show-template").html())(context);
             };
 
             DiscussionThreadShowView.prototype.render = function() {
-                this.$el.html(this.renderTemplate());
+                edx.HtmlUtils.setHtml(
+                    this.$el,
+                    this.renderTemplate()
+                );
                 this.delegateEvents();
                 this.renderAttrs();
                 this.$("span.timeago").timeago();
                 this.convertMath();
-                this.highlight(this.$(".post-body"));
-                this.highlight(this.$("h1,h3"));
+                this.$(".post-body");
+                this.$("h1,h3");
                 return this;
             };
 
             DiscussionThreadShowView.prototype.convertMath = function() {
-                var element;
-                element = this.$(".post-body");
-                element.html(DiscussionUtil.postMathJaxProcessor(DiscussionUtil.markdownWithHighlight(element.text())));
```

Comment:
@robrap current version makes more sense than `alreadyEsscaped` (which also had a bug in - profile page was broken). Yes, I'm more happy with it, than previous; ideally it should also preserve tags in post body itself, but as I've already said looks like it is "by design" decision in markdown converter + it is existing behavior